Source/CarthageKit/Xcode.swift

```diff
@@ -1080,85 +1080,152 @@ public typealias BuildSchemeProducer = SignalProducer<TaskEvent<(ProjectLocator,
 ///
 /// Returns producers in the same format as buildInDirectory().
 public func buildDependencyProject(dependency: ProjectIdentifier, _ rootDirectoryURL: NSURL, withOptions options: BuildOptions, sdkFilter: SDKFilterCallback = { .Success($0.0) }) -> SignalProducer<BuildSchemeProducer, CarthageError> {
-	let rootBinariesURL = rootDirectoryURL.URLByAppendingPathComponent(CarthageBinariesFolderPath, isDirectory: true).URLByResolvingSymlinksInPath!
 	let rawDependencyURL = rootDirectoryURL.URLByAppendingPathComponent(dependency.relativePath, isDirectory: true)
 	let dependencyURL = rawDependencyURL.URLByResolvingSymlinksInPath!
 
 	let schemeProducers = buildInDirectory(dependencyURL, withOptions: options, sdkFilter: sdkFilter)
-	return SignalProducer.attempt { () -> Result<SignalProducer<BuildSchemeProducer, CarthageError>, CarthageError> in
+	return symlinkBuildPathForDependencyProject(dependency, rootDirectoryURL: rootDirectoryURL)
+		.concat(symlinkCheckoutPathForDependencyProject(dependency, rootDirectoryURL: rootDirectoryURL))
+		.flatMap(.Merge) { _ -> SignalProducer<BuildSchemeProducer, CarthageError> in
+			return schemeProducers
+				.mapError { error in
+					switch (dependency, error) {
+					case let (_, .NoSharedFrameworkSchemes(_, platforms)):
+						return .NoSharedFrameworkSchemes(dependency, platforms)
+
+					case let (.GitHub(repo), .NoSharedSchemes(project, _)):
+						return .NoSharedSchemes(project, repo)
+
+					default:
+						return error
+					}
+			}
+		}
+}
+
+/// Creates symlink between the dependency build folder and the root build folder
+///
+/// Returns a signal indicating success
+private func symlinkBuildPathForDependencyProject(dependency: ProjectIdentifier, rootDirectoryURL: NSURL) -> SignalProducer<Bool, CarthageError> {
+	return SignalProducer.attempt {
+		let rootBinariesURL = rootDirectoryURL.URLByAppendingPathComponent(CarthageBinariesFolderPath, isDirectory: true).URLByResolvingSymlinksInPath!
+		let rawDependencyURL = rootDirectoryURL.URLByAppendingPathComponent(dependency.relativePath, isDirectory: true)
+		let dependencyURL = rawDependencyURL.URLByResolvingSymlinksInPath!
+
+		do {
+			try NSFileManager.defaultManager().createDirectoryAtURL(rootBinariesURL, withIntermediateDirectories: true, attributes: nil)
```

Comment:
I guess you just moved this code. 😄 But if you don't updating that, it'd be appreciated!